### DISCUSSION TITLE
Help with chromecast

### DISCUSSION THREAD

**Discussion by reboot89** [Mygod/VPNHotspot#405] [Q&A]
Hi, first of all thanks for this wonderful app.
I really need help setting it up tho, wish there was a video or a guide explaining how to make it work with a chromecast.

So i have rooted my android phone only cause i did find your hotspot app, and it would be wonderful if i could make it work with my chromecast.

I have tried using hotspot shield vpn the paid one, connected to one of their servers, then on your app on my rooted phone thicked the wi-fi hotspot and also the swlan0,  then did a factory reset on the chromecast in order for it to connect to the wi-fi hotspot from your app.

So here comes the part where i get stuck and confused, as you know when setting up the chromecast from the Google home app, one have to connect to a wi-fi in order for it to work, but on the same phone that i did create a hotspot, the Google home app when selecting a wi-fi wont display/show the hotspot from your app on the same phone, so i got stuck and confused by that part, and figured out to create the hotspot from your app with my newly rooted phone, then use a other device such as my android box to get the Google home app and configure it to set it up with the chromecast as it was able for another device to find the hotspot from your app setup by my phone. 

So it did the job done, and i succusfully connected to the wi-fi hotspot on the chromecast, BUT... whenever i try to use my phone to chromecast a YouTube video or other streaming apps, it shows the chromecast button and my room which i have configued with the chromecast it self,
The chromecast button when i tap it, it would try and search but aften 5 second, NO success no matter how many times i tap the button.

Sorry for the long massage and my broken english, but if you know how to fix or have any solutions, please do try and help me understand how to make it work somehow.

Again thanks a lot for all the efforts, your work and help is all appriciated. Also i am using the latest version of your app downloaded here from your github.

**Comment by GaryThurman** [comment]
Maybe you'll find your answer here
https://forum.xda-developers.com/t/app-root-5-0-vpn-hotspot-vpn-tethering-wi-fi-repeater-open-source.3741604/

**Comment by Mygod** [comment]
From my experience (a long time ago), casting is not supported from the device that is broadcasting the repeater. I am not aware of a workaround. Meanwhile, you could simply cast from another device.

**Comment by Massimo-B** [comment]
Coing from  #449...
When using a Chromecast connected to the Tethered Wifi of an Android device with mobile uplink, the casting is supported and does work. It's just broken by enabling a VPN and here VPNHotspot comes in to solve that. It already solves that the tethered Wifi is using the VPN, which is its main purpose, but finding and controlling a local Chromecast (via Multicast DNS?) still does not work. The casting should happen in the VPN network (if possible), not the local one.

Finally with that setup using VPN I would expect to see all local and remote Chromecast devices.

**Comment by Massimo-B** [comment]
Years later I still struggle with this issue.
Meanwhile on OxygenOS 16, still rooting for things like this, cast device today is a Xiaomi Mi TV Stick 4k with Android TV, which can stream stand-alone but also serve as a Chromecast device.

Established OpenVPN, started the Wifi hotspot, then enabled the new wlan2 in the _VPN Hotspot_ app to make connected clients being routed via the VPN.
That works and also the connected Mi TV Stick 4k is streaming via VPN. But cast does not work with that single Android device.
Only workarounds are either turning off the VPN or using a second Android device in the tethered (VPN) network.

I know I still could use the stick in standalone mode which the original old chromecast can't, but often I also need the cast mode for sending specific streams via UPnP. 
This combination of a single Android device with mobile or also open Wifi uplink AND VPN I often need when traveling in other countries.

I'm still convinced getting a Chromecast connecting to the VPN is one of the main purposes of this app as described in the summary.
